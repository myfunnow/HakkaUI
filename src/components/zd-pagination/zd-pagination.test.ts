
import { describe, it, expect, beforeAll } from "vitest"
import { mount } from '@vue/test-utils'
import ZdPagination from './zd-pagination.vue'

beforeAll(() => {
  global.ResizeObserver = class ResizeObserver {
    observe() { }
    unobserve() { }
    disconnect() { }
  }
})

describe('Pagination', async () => {
  expect(ZdPagination).toBeTruthy()

  const wrapper = await mount(ZdPagination, {
    props: {
      visible: 5,
      page: 4,
      pageSize: 9,
      totalCount: 50
    },
  })

  it('should to match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should contain the correct text', () => {
    expect(wrapper.text()).toContain('第 28 - 36，共 50 筆結果')
  })

  it('should display the correct page action button', () => {
    expect(wrapper.text()).toContain('1456')
  })
})
