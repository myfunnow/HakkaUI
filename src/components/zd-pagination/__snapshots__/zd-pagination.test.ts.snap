// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > should to match the snapshot 1`] = `
"<div data-v-49ad08a6="" class="zd-pagination flex flex-col justify-center">
  <p data-v-49ad08a6="" class="text mb-0 text-center text-sm text-gray-80">第 28 - 36，共 50 筆結果</p>
  <div data-v-49ad08a6="" class="n-pagination flex justify-center leading-20px mt-5" style="--n-prefix-margin: 0 8px 0 0; --n-suffix-margin: 0 0 0 8px; --n-item-font-size: 14px; --n-select-width: unset; --n-select-margin: 0 0 0 8px; --n-input-width: 60px; --n-input-margin: 0 0 0 8px; --n-input-margin-rtl: 0 8px 0 0; --n-item-size: 28px; --n-item-text-color: rgb(51, 54, 57); --n-item-text-color-disabled: rgba(194, 194, 194, 1); --n-item-text-color-hover: #36ad6a; --n-item-text-color-active: #18a058; --n-item-text-color-pressed: #0c7a43; --n-item-color: #0000; --n-item-color-hover: #0000; --n-item-color-disabled: rgb(250, 250, 252); --n-item-color-active: #0000; --n-item-color-active-hover: #0000; --n-item-color-pressed: #0000; --n-item-border: 1px solid #0000; --n-item-border-hover: 1px solid #0000; --n-item-border-disabled: 1px solid rgb(224, 224, 230); --n-item-border-active: 1px solid #18a058; --n-item-border-pressed: 1px solid #0000; --n-item-padding: 0 4px; --n-item-border-radius: 3px; --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-jumper-font-size: 14px; --n-jumper-text-color: rgb(51, 54, 57); --n-jumper-text-color-disabled: rgba(194, 194, 194, 1); --n-item-margin: 0 0 0 8px; --n-item-margin-rtl: 0 8px 0 0; --n-button-icon-size: 16px; --n-button-icon-color: rgb(51, 54, 57); --n-button-icon-color-hover: rgb(51, 54, 57); --n-button-icon-color-pressed: rgb(51, 54, 57); --n-button-color-hover: #0000; --n-button-color: #0000; --n-button-color-pressed: #0000; --n-button-border: 1px solid rgb(224, 224, 230); --n-button-border-hover: 1px solid rgb(224, 224, 230); --n-button-border-pressed: 1px solid rgb(224, 224, 230);">
    <!---->
    <div class="n-pagination-item">
      <div data-v-49ad08a6="" class="zd-icon flex items-center justify-center transition-opacity w-4 h-4"><svg viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
          <path fill-rule="evenodd" clip-rule="evenodd" d="M15.325 3.27504C15.6262 3.50098 15.7091 3.91003 15.535 4.2315L15.475 4.32504L9.925 11.725C9.87167 11.7962 9.861 11.8886 9.893 11.9682L9.925 12.025L15.475 19.4251C15.7235 19.7565 15.6564 20.2266 15.325 20.4751C15.0238 20.7011 14.6079 20.6661 14.348 20.4089L14.275 20.3251L8.72499 12.925C8.29167 12.3473 8.26072 11.5683 8.63214 10.9618L8.725 10.825L14.275 3.42504C14.5235 3.09367 14.9936 3.02652 15.325 3.27504Z" fill="currentColor"></path>
        </svg></div>
    </div>
    <div class="n-pagination-item n-pagination-item--clickable">1</div>
    <!---->
    <div class="n-pagination-item"><i class="n-base-icon"><svg viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg">
          <g stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
            <g fill="currentColor" fill-rule="nonzero">
              <path d="M4,7 C4.55228,7 5,7.44772 5,8 C5,8.55229 4.55228,9 4,9 C3.44772,9 3,8.55229 3,8 C3,7.44772 3.44772,7 4,7 Z M8,7 C8.55229,7 9,7.44772 9,8 C9,8.55229 8.55229,9 8,9 C7.44772,9 7,8.55229 7,8 C7,7.44772 7.44772,7 8,7 Z M12,7 C12.5523,7 13,7.44772 13,8 C13,8.55229 12.5523,9 12,9 C11.4477,9 11,8.55229 11,8 C11,7.44772 11.4477,7 12,7 Z"></path>
            </g>
          </g>
        </svg></i></div>
    <!---->
    <div class="n-pagination-item n-pagination-item--active n-pagination-item--clickable">4</div>
    <!---->
    <div class="n-pagination-item n-pagination-item--clickable">5</div>
    <!---->
    <div class="n-pagination-item n-pagination-item--clickable">6</div>
    <div class="n-pagination-item">
      <div data-v-49ad08a6="" class="zd-icon flex items-center justify-center transition-opacity w-4 h-4"><svg viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
          <path fill-rule="evenodd" clip-rule="evenodd" d="M8.67504 3.27504C8.3738 3.50098 8.29091 3.91003 8.46505 4.2315L8.52504 4.32504L14.075 11.725C14.1284 11.7962 14.139 11.8886 14.107 11.9682L14.075 12.025L8.52504 19.4251C8.27652 19.7565 8.34368 20.2266 8.67505 20.4751C8.97629 20.7011 9.39219 20.6661 9.65204 20.4089L9.72505 20.3251L15.2751 12.925C15.7084 12.3473 15.7393 11.5683 15.3679 10.9618L15.275 10.825L9.72504 3.42504C9.47652 3.09367 9.00642 3.02652 8.67504 3.27504Z" fill="currentColor"></path>
        </svg></div>
    </div>
    <!---->
    <!---->
    <!---->
  </div>
</div>"
`;
